### PR TITLE
Website 11719/implement 404 page

### PR DESCRIPTION
**Pull request by GabrielEger2**
### Ticket(s)

- [_[11719]_](https://airtable.com/appfJZShN8K4tcWHU/pagHwvtNNoIIuXqAz?GOswY=rec8QTdiE9nKgExRW)

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Description

This task involves researching, designing, and implementing a custom 404 Not Found page for our website. The page should align with CV's visual identity and provide a helpful user experience for visitors who reach a non-existent page on our site.

#### Please include a screenshot of the Figma design file
image
image

#### Before
image
image

#### After
image
image

### Review your work
Please self review your work before requesting further review. _Wait until you reach the next screen to check off the boxes._

#### Preview
Use your local build.

- [x] Everything works as expected
- [x] Check changed page(s)
- [x] Compare figma screenshot with the changes
- [x] Compare linked Figma section to changes made
- [x] Refresh the page, check that using the go-back button works as expected
- [x] Check large screen size (>1400)
- [x] Check tablet screen size (<786)
- [x] Check mobile screen size (<375)
- [x] Check that links work
- [x] Check for any uncaught ESLint problems

#### Code Clarity and Readability
- [x] Are variable and function names descriptive, meaningful, in the appropriate case (ex: snake, camel)?
- [x] Is the code self-explanatory and easy to understand?

#### Code Efficiency
- [x] Is the code DRY?

#### Code Modularity and Reusability
- [x] Is the code and any new components organized in a logical location in the directory?

#### Best Practices and Coding Standards
- [x] Does the code follow the established coding standards and style guide of the project?
- [x] Are there any violations of best practices that should be addressed?
- [x] Are all files TypeScript?

### Overall PR Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings in the console
- [x] There are no new linting errors/problems in my code
- [x] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand

### CODE REVIEW

File: content/404.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import {
+  Header, MetaPageHead,
+} from './content.types.ts';
+
+    interface NotFoundPageContent {
+      meta: MetaPageHead
+      page: Header;
+    }
```
Comment (LeoFuna) [inline content/404.ts:8]: Hey @GabrielEger2! Seems we have some extra spaces here.

**Review by LeoFuna — APPROVED** [review]
Great job! Just a format adjust and we are good to go. :handshake:

File: content/404.ts (hunk 1 above)
Comment (e-papanicolas) [reply content/404.ts:8]: missing a semi colon on line 6

File: pages/404.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import React from 'react';
+
+import externalLinks from '../components/functional/ExternalLinks.tsx';
+import IndividualPageHead from '../components/helper/IndividualPageHead.tsx';
+import ImageContainer from '../components/layout/ImageContainer.tsx';
+import SectionContainer from '../components/layout/SectionContainer.tsx';
+import content from '../content/404.ts';
+
+export default function PageNotFound() {
+  externalLinks();
```
Comment (e-papanicolas) [inline pages/404.tsx:11]: not sure if we need this since there are no external links on the page.

File: content/404.ts (hunk 1 above)
Comment (GabrielEger2) [reply content/404.ts:8]: That's odd, I'm pretty sure I had ESLint turned on. I'll fix it ASAP

**Review by e-papanicolas — APPROVED** [review]

**Review by LeoFuna — APPROVED** [review]

**Pull request merged**